server/routes/ad.ts

```diff
@@ -398,51 +398,81 @@ const getAnomalyResults = async (
   try {
     const {
       from = 0,
-      size = 20,
+      size = 30,
       sortDirection = SORT_DIRECTION.DESC,
-      sortField = 'startTime',
+      sortField = AD_DOC_FIELDS.DATA_START_TIME,
+      range = undefined,
       //@ts-ignore
-    } = req.query as DetectorResultsQueryParams;
+    } = req.query as {
+      from: number;
+      size: number;
+      sortDirection: SORT_DIRECTION;
+      sortField?: string;
+      range?: any;
+    };
     const { detectorId } = req.params;
 
     //Allowed sorting columns
     const sortQueryMap = {
       anomalyGrade: { anomaly_grade: sortDirection },
       confidence: { confidence: sortDirection },
-      startTime: { data_start_time: sortDirection },
-      endTime: { data_end_time: sortDirection },
+      [AD_DOC_FIELDS.DATA_START_TIME]: {
+        [AD_DOC_FIELDS.DATA_START_TIME]: sortDirection,
+      },
+      [AD_DOC_FIELDS.DATA_END_TIME]: {
+        [AD_DOC_FIELDS.DATA_END_TIME]: sortDirection,
+      },
     } as { [key: string]: object };
     let sort = {};
     const sortQuery = sortQueryMap[sortField];
     if (sortQuery) {
       sort = sortQuery;
     }
+
+    let rangeObj = range;
+
+    if (range !== undefined && typeof range === 'string') {
+      rangeObj = JSON.parse(range);
```

Comment:
Are we sure this will be always valid JSON string ? If not then wrap this inside `try...catch` this will fail JSON.parse("hello")

Reply:
It is already inside `try catch`, and it does log the error in case of any exception and then
```
return { ok: false, error: err.message }
```
The question is: given `range` is not undefined, should we fail the request, if `range` is invalid string? If yes, then we may keep current implementation; if no, aka we just ignore invalid `range` string. I would probably go with `Failing the request` for now, because `range` is specified, but invalid input, and we should fail it to let caller know the reason, instead of swallowing issue and returning result caller doesn't expect.